internal/testing/testsdk/providerserver/providerserver.go

```diff
@@ -202,6 +239,21 @@ func (s ProviderServer) ApplyResourceChange(ctx context.Context, req *tfprotov6.
 		}
 
 		resp.NewState = newState
+		newIdentity = updateResp.NewIdentity
+	}
+
+	if newIdentity != nil {
+		newIdentity, diag := IdentityValuetoDynamicValue(identitySchemaResp.Schema, *newIdentity)
```

Comment:
```suggestion
		newIdentityVal, diag := IdentityValuetoDynamicValue(identitySchemaResp.Schema, *newIdentity)
```

Suggestion: a new variable name makes it a bit easier to read, since we've declare `newIdentity` as a `Value` and then exchange it for another type (`DynamicValue`).